src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs

```diff
@@ -3954,6 +3954,48 @@
     ("SveVecPairBinOpTest.template",      new Dictionary<string, string> { ["TestName"] = "SveTransposeOdd_uint",                                                                                  ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "TransposeOdd",                                                         ["RetVectorType"] = "Vector",    ["RetBaseType"] = "UInt32",  ["Op1VectorType"] = "Vector",    ["Op1BaseType"] = "UInt32", ["Op2VectorType"] = "Vector",    ["Op2BaseType"] = "UInt32",                                                              ["LargestVectorSize"] = "64", ["NextValueOp1"] = "TestLibrary.Generator.GetUInt32()",        ["NextValueOp2"] = "TestLibrary.Generator.GetUInt32()",                                                                                        ["ValidateEntry"] = "result[i] != left[index * 2 + 1] || result[i + 1] != right[index * 2 + 1]"}),
     ("SveVecPairBinOpTest.template",      new Dictionary<string, string> { ["TestName"] = "SveTransposeOdd_ulong",                                                                                 ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "TransposeOdd",                                                         ["RetVectorType"] = "Vector",    ["RetBaseType"] = "UInt64",  ["Op1VectorType"] = "Vector",    ["Op1BaseType"] = "UInt64", ["Op2VectorType"] = "Vector",    ["Op2BaseType"] = "UInt64",                                                              ["LargestVectorSize"] = "64", ["NextValueOp1"] = "TestLibrary.Generator.GetUInt64()",        ["NextValueOp2"] = "TestLibrary.Generator.GetUInt64()",                                                                                        ["ValidateEntry"] = "result[i] != left[index * 2 + 1] || result[i + 1] != right[index * 2 + 1]"}),
+
+    ("SveTestTest.template",      new Dictionary<string, string> { ["TestName"] = "SveTestAnyTrue_sbyte",                                                                                ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "TestAnyTrue",                                                        ["MaskBaseType"] = "SByte",   ["Op1Value"] = "Sve.CreateTrueMaskSByte(SveMaskPattern.All)",  ["Op2Value"] = "Sve.CreateTrueMaskSByte(SveMaskPattern.All)",                        ["ValidateEntry"] = "result != true"}),
```

Comment:
To make sure we are testing this correctly, I think we should just use `GetRandomByte()` to populate the operands and check if it returns the expected result.

Reply:
It wouldn't be trivial to verify this if the inputs are random.
The op values I have are good enough to make sure we are doing the right thing for all 3 Test APIs.

To demonstrate, let's look at the lowering code:
```cpp
        case NI_SveTestAnyTrue
        {
            LowerNodeCC(node, GenCondition::NE);
            node->gtType = TYP_VOID;
            return node->gtNext;
        }

        case NI_Sve_TestFirstTrue:
        {
            LowerNodeCC(node, GenCondition::SLT);
            node->gtType = TYP_VOID;
            return node->gtNext;
        }

        case NI_Sve_TestLastTrue:
        {
            LowerNodeCC(node, GenCondition::ULT);
            node->gtType = TYP_VOID;
            return node->gtNext;
        }
```
These **are** the correct conditions -> [see GCC implementation](https://github.com/gcc-mirror/gcc/blob/7c6f79eea9febce3b21c5783bac9b0a36e08f003/gcc/config/aarch64/aarch64-sve-builtins-base.cc#L3164).
Now, if I change any of those `LowerNodeCC` conditions to something other than what it currently is, the tests will fail.

Reply:
For testing the mask vectors should be filled randomly using `Helpers.getMaskByte()` etc

Then for validation (duplicated for every required type...):

```csharp

// Find any occurrence where both left and right and set
bool TestAnyTrue(byte left[], byte right[])
{
   for (int i=0; i< left.Length)
   {
      if (left[i]>0 && right[i]>0) return true;
   }
   return false
}

// Find first true value in left, and return contents of right
bool TestFirstTrue(byte left[], byte right[])
{
   for (int i=0; i< left.Length)
   {
      if (left[i]>0) return right[i]>0 ? true : false
   }
   return false
}

// Find last true value in left, and return contents of right
bool TestLastTrue(byte left[], byte right[])
{
   for (i = left.Length-1; i>0; i--)
   {
      if (left[i]>0) return right[i]>0 ? true : false
   }
   return false
}
```

Reply:
I see, so we basically simulate what the API does?